frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js

```diff
@@ -126,85 +152,121 @@ const AdditionalParamLabel = styled.span`
 export default function AnalysisParamForm({
   inputVals,
   errorMessages,
-  checked,
-  setChecked,
+  checkedWindow,
+  setCheckedWindow,
   setAnalysisParams,
+  checkedAdvanced,
+  setCheckedAdvanced,
   paramErrors,
   setParamErrors,
   analysisParams,
 }) {
   const updateParams = (newParams) => {
     const updatedParams = { ...analysisParams, ...newParams };
 
-    if (newParams.twitchWidths !== undefined) {
+    if (newParams.twitchWidths) {
       validateTwitchWidths(updatedParams);
     }
-    if (newParams.startTime !== undefined || newParams.endTime !== undefined) {
+    if (newParams.startTime || newParams.endTime) {
       // need to validate start and end time together
       validateWindowBounds(updatedParams);
     }
-    if (newParams.prominenceFactor !== undefined) {
-      validateProminenceFactor(updatedParams);
+    if (newParams.prominenceFactorPeaks) {
+      validateProminenceFactorMin(updatedParams);
+    }
+    if (newParams.prominenceFactorValleys) {
+      validateProminenceFactorMax(updatedParams);
+    }
+    if (newParams.widthFactorPeaks) {
+      validateWidthFactorMin(updatedParams);
     }
-    if (newParams.widthFactor !== undefined) {
-      validateWidthFactor(updatedParams)
+    if (newParams.widthFactorValleys) {
+      validateWidthFactorMax(updatedParams);
     }
     setAnalysisParams(updatedParams);
   };
 
-  const validateTuple = (userInput) => {
-    const tuple = userInput.split(",")
-    //check there are only two entrees
-    if (tuple.length !== 2) {
-      return false
-    }
-    //check both entrees are numbers
-    try {
-      parseFloat(tuple[0])
-      parseFloat(tuple[1])
-    } catch (e) {
-      return false
+  const isValidPositiveNumber = (value) => {
+    return +value > 0;
+  };
+
+  const validateProminenceFactorMin = (updatedParams) => {
```

Comment:
These 4 validation functions are almost identical, you should  just make one function and just pass in the field you want to validate